spec/ruby-push-notifications/fcm/fcm_notification_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RubyPushNotifications
+  module FCM
+    describe FCMNotification do
+
+      let(:registration_ids) { %w[a] }
+      let(:data) {
+        {
+          a: {
+            'notification': {
+              'title': 'Greetings Test',
+              'body': 'Remember to test! ALOT!',
+              'forceStart': 1,
+              'sound': 'default',
+              'vibrate': 'true',
+              'icon': 'fcm_push_icon'
+            },
+            'android': {
+              'priority': 'high',
+              'vibrate': 'true'
+            },
+            'data': {
+              'url': 'https://www.google.com'
+            },
+            'webpush': {
+              'headers': {
+                'TTL': '60'
+              }
+            },
+            'priority': 'high'
+          }
+        }
+      }
+      let(:notification) {
+        build :fcm_notification,
+              registration_ids: registration_ids,
+              data: data
+      }
+
+      it 'builds the right fcm json' do
+        expect(notification.as_fcm_json).to eq JSON.dump({registration_ids: @registration_ids}.merge(data))
```

Comment:
Ok, the test failing is this one. And it makes sense, as here you're expecting the result of `notification.as_fcm_json` to include the `registration_ids` key, but looking at the code of the method `FCMNotification#as_fcm_json`, it doesn't add those values. Either of them is wrong, which one depends on wether the `registration_ids` field is actually required as part of the json or not

Reply:
Thank you for shedding light on this. As i was not the original author i had some issues deciphering the code while i was writing the tests.

I have resolved this and i will push the changes. I just need to check remaining hound-related changes that is required.